numpy/core/src/common/ufunc_override.c

```diff
@@ -37,7 +31,7 @@ get_non_default_array_ufunc(PyObject *obj)
     }
 
     /* Fast return for ndarray */
-    if ((PyObject *)Py_TYPE(obj) == ndarray) {
+    if (PyArray_CheckExact(obj)) {
```

Comment:
@mattip: If you're changing this, you can now use `(PyObject *)&PyArray_Type` above too, right?

Reply:
This made sense to me, but a trial gave an immediate segmentation fault...

Reply:
Solved with `#define _MULTIARRAYMODULE` above